elastic/kibana
# [Synthetics] Adds additional heartbeat configuration options to the Synthetics app

## Summary

Closes https://github.com/elastic/kibana/issues/140864
Relates to https://github.com/elastic/synthetics-dev/issues/159
Relates to https://github.com/elastic/synthetics/issues/655

Adds missing heartbeat configs to the Synthetics app. The additional configs are specified below:

HTTP (proxy_headers, mode, ipv4, ipv6, response.include_body_max_bytes, check.response.json)
![image](https://user-images.githubusercontent.com/11356435/231040397-4f80d3af-ff08-4ef1-bbd9-c49f63d29bd0.png)
![image](https://user-images.githubusercontent.com/11356435/231040371-b7baa693-d573-46e1-b0f1-6d21c32522b8.png)


TCP (mode, ipv4, ivp6)
![image](https://user-images.githubusercontent.com/11356435/231040065-ad865160-7f6c-4450-ab1c-98c17aedd3f0.png)

ICMP (mode, ipv4, ipv6)
![image](https://user-images.githubusercontent.com/11356435/231039925-d2a2e9ab-69aa-4d74-8c3e-91223dd963d1.png)


### Testing

1. Create a private location
2. Create an http monitor selecting both a public and private location, adjusting the settings for mode, ipv4, ipv6, response.include_body_max_bytes, check.response.json and proxy headers. 
3. Navigate to the edit page for that monitor, ensure the configuration above was saved to the monitor settings
4. Create an icmp monitor selecting both a public and private location, adjusting the settings for mode, ipv6, and ipv4.
5. Navigate to the dit page for that monitor, ensure the configuration above was saved to the monitor settings
6. Create an tcp monitor selecting both a public and private location, adjusting the settings for mode, ipv6, and ipv4.
7. Navigate to the dit page for that monitor, ensure the configuration above was saved to the monitor settings
8. Navigate to the agent policy for the private location selected. Ensure that the configuration options are represented on the the individual integration policies
